code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/wrappedform/WrappedFormFieldForm.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+import {Form, models, scout} from '@eclipse-scout/core';
+import WrappedFormFieldFormModel from './WrappedFormFieldFormModel';
+
+export default class WrappedFormFieldForm extends Form {
+
+  constructor() {
+    super();
+  }
+
+  _jsonModel() {
+    return models.get(WrappedFormFieldFormModel);
+  }
+
+  _init(model) {
+    super._init(model);
+
+    this.innerFormField = this.widget('InnerFormField');
+    this.closeInnerFormButton = this.widget('CloseInnerFormButton');
+    this.wrappedFormField = this.widget('WrappedFormField');
+
+    this.innerFormField.on('propertyChange:value', event => {
+      let innerForm = null;
+      if (event.newValue) {
+        innerForm = scout.create(event.newValue, {parent: this.wrappedFormField});
+      }
+      this.wrappedFormField.setInnerForm(innerForm);
+    });
+    this.wrappedFormField.on('propertyChange:innerForm', event => this.closeInnerFormButton.setEnabled(!!event.newValue));
+    this.closeInnerFormButton.on('click', event => {
+      this.wrappedFormField.innerForm.close();
+      this.innerFormField.setValue(null);
```

Comment:
Is setting to null necessary? This should be done automatically when form is closed (destroyed)

Reply:
It is necessary, the innerFormField's selected lookupRow is not reseted to null anywhere else.
But it is not necessary to close the wrappedFormField's innerForm, this is done by the innerFormField's propertyChange-listener.